src/Angular.js

```diff
@@ -840,43 +842,48 @@ function copy(source, destination) {
 
     stackSource.push(source);
     stackDest.push(destination);
-    return copyRecurse(source, destination);
+    return copyRecurse(source, destination, currentDepth);
   }
 
-  return copyElement(source);
+  return copyElement(source, currentDepth);
+
+  function copyRecurse(source, destination, currentDepth) {
+    currentDepth++;
+    if (maxDepth !== -1 && currentDepth > maxDepth) {
```

Comment:
By setting `maxDepth` to `NaN` when not specified, this condition could be: `if (currentDepth > maxDepth)`